src/controller/c/lidar.c

```diff
@@ -391,6 +390,7 @@ double wb_lidar_get_max_range(WbDeviceTag tag) {
 
 const float *wb_lidar_get_range_image(WbDeviceTag tag) {
   AbstractCamera *ac = lidar_get_abstract_camera_struct(tag);
+  robot_mutex_lock_step();
```

Comment:
I would move this call to `robot_mutex_lock_step()` down as I fear it could lock the mutex and return (if invalid device tag or remote control mode) without unlocking it...

Reply:
Done for all devices